.github/workflows/opensource-projects.yml

```diff
@@ -0,0 +1,192 @@
+# Copyright 2025, Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+
+name: Build opensource projects
+
+permissions: read-all
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - '*.md'
+      - 'snap/**'
+      - 'benchmarks/**'
+      - 'tests/**'
+      - 'examples/**'
+      - 'tools/**'
+      - 'utils/**'
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-ispc:
+    name: Build ISPC
```

Comment:
does it make sense to build ispc once and reuse within workflows? I believe it is possible.
you have the same build for https://github.com/ispc/ispc/blob/main/.github/workflows/openmoonray.yml, for example

Reply:
It's a good idea but I think it's out of the scope of this PR. It can be done separately.